examples/notebooks/beam-ml/image_processing_beam.ipynb

```diff
@@ -0,0 +1,933 @@
+      ],
+      "metadata": {
+        "id": "bfgy0Z_gX_lH"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "P2oPvZkbfEPo"
+      },
+      "outputs": [],
+      "source": [
+        "grayscale = []\n",
+        "for i in x_train:\n",
+        "  grayImage = 0.07 * i[:,:,2] + 0.72 * i[:,:,1] + 0.21 * i[:,:,0]\n",
+        "  grayscale.append(grayImage)\n",
+        "x_train_gray = np.asarray(grayscale)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jzQ2Zulg99NU"
+      },
+      "source": [
+        "## Defining DoFns for Image Preprocessing\n",
+        "\n",
+        "[DoFn](https://beam.apache.org/releases/typedoc/current/interfaces/transforms_pardo.DoFn) stands for \"Do Function\". In Apache Beam, it is a set of operations that can be applied to individual elements of a PCollection (a collection of data). It is similar to a function in Python, except that it is used in Beam Pipelines to apply various transformations. DoFns can be used in various Apache Beam transforms, such as ParDo, Map, Filter, and FlatMap."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "cqm-m0cONsZS"
+      },
+      "outputs": [],
+      "source": [
+        "class StandardizeImage(beam.DoFn):\n",
+        "  def process(self, element: np.ndarray):\n",
+        "    element = element/255.0\n",
+        "    return [element]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "mZhFCgPxPEwm"
+      },
+      "outputs": [],
+      "source": [
+        "class NormalizeImage(beam.DoFn):\n",
+        "  def process(self, element: np.ndarray):\n",
+        "    element = (element-element.min())/(element.max()-element.min())\n",
+        "    return [element]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "gv23KPt5NyXT"
+      },
+      "outputs": [],
+      "source": [
+        "class GrayscaleImage(beam.DoFn):\n",
+        "  def process(self, element: np.ndarray):\n",
+        "    element = 0.07 * element[:,:,2] + 0.72 * element[:,:,1] + 0.21 * element[:,:,0]\n",
+        "    return [element]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8gz7_SvN-P2L"
+      },
+      "source": [
+        "## Training a Convolutional Neural Network\n",
+        "\n",
+        "A Convolutional Neural Network (CNN) is one of the most popular model for image processing. Here is a brief description of the convolutional layers used in the model.\n",
+        "* **Reshape**: Changes the shape of the input data to the desired size. <br>\n",
```

Comment:
The bullet, line break format here creates weird formatting on GitHub. Could we just get rid of the `<br>` tags?